## Day 1

W400 free, heats and finals 
M400 free, heats and finals

W100Fly heats and semis
M100Breast heats and semis

4x100 free relay M and W (still need to parse the results)


Women's 400 Free final, dubbed as race of the century, with Katie Ledecky, Ariane Titmus, and Summer McIntosh featured. Titmus won the gold, McIntosh the sliver, and Ledecky the bronze. It was Ledeky's 11th  overall Olympic medal, making her the most decorated female swimmer Olympic history.






In [1]:
from utils import parse_100_heats, parse_400_heats, parse_800_heats, sort_heats
from pathlib import Path

outdir = Path('ParsedData')

w100fly_heats_df=parse_100_heats('W100Fly_heats_results.csv', 'Womens 100M Butterfly', 'Heats')
w400free_heats_df=parse_400_heats('W400Free_heats_results.csv', 'Womens 400M Freestyle', 'Heats')

m100Breast_heats_df=parse_100_heats('M100Breast_heats_results.csv', 'Mens 100M Breaststroke', 'Heats')
m400free_heats_df=parse_400_heats('M400Free_heats_results.csv', 'Mens 400M Freestyle', 'Heats')

#write to csv files in ParsedData directory

w100fly_heats_df.to_csv(outdir / 'W100Fly_heats_results.csv', index=False)
w400free_heats_df.to_csv(outdir / 'W400Free_heats_results.csv', index=False)
m100Breast_heats_df.to_csv(outdir / 'M100Breast_heats_results.csv', index=False)
m400free_heats_df.to_csv(outdir / 'M400Free_heats_results.csv', index=False)

display(w100fly_heats_df.head())   
display(w400free_heats_df.head())
display(m100Breast_heats_df.head())
display(m400free_heats_df.head())


Place Country            Name Age    RT     Status   Time 50 Split  \
0     1     CHN     Yufei ZHANG  26  0.70  QUALIFIED  56.50    26.15   
1     2     JPN    Mizuki HIRAI  17  0.57  QUALIFIED  56.71    26.54   
2     3     USA     Torri HUSKE  21  0.64  QUALIFIED  56.72    26.01   
3     4     USA  Gretchen WALSH  21  0.77  QUALIFIED  56.75    25.98   
4     5     AUS     Emma MCKEON  30  0.70  QUALIFIED  56.79    26.14   

  100 Split                  Event  Round  
0     30.35  Womens 100M Butterfly  Heats  
1     30.17  Womens 100M Butterfly  Heats  
2     30.71  Womens 100M Butterfly  Heats  
3     30.77  Womens 100M Butterfly  Heats  
4     30.65  Womens 100M Butterfly  Heats

Place Country               Name Age    RT     Status     Time    50m  \
0     1     USA      Katie LEDECKY  27  0.67  QUALIFIED  4:02.19  27.82   
1     2     AUS     Ariarne TITMUS  23  0.72  QUALIFIED  4:02.46  27.43   
2     3     NZL  Erika FAIRWEATHER  20  0.74  QUALIFIED  4:02.55  27.67   
3     4     CAN    Summer MCINTOSH  17  0.76  QUALIFIED  4:02.65  27.68   
4     5     AUS  Jamie Ann PERKINS  19  0.76  QUALIFIED  4:03.30  28.34   

    100m   150m   200m   250m   300m   350m   400m                  Event  \
0  30.05  30.48  30.71  30.91  31.10  30.96  30.16  Womens 400M Freestyle   
1  29.67  30.39  30.67  31.01  31.26  31.60  30.43  Womens 400M Freestyle   
2  30.00  30.55  30.89  30.72  31.25  31.03  30.44  Womens 400M Freestyle   
3  30.09  30.74  30.91  30.81  31.25  31.11  30.06  Womens 400M Freestyle   
4  30.21  30.65  31.20  31.07  31.26  30.90  29.67  Womens 400M Freestyle   

   Round  
0  Heats  
1  Heats  
2  Heats  
3  Heats  
4  Heats

Place Country                Name Age    RT     Status   Time 50 Split  \
0     1     NED      Caspar CORBEAU  23  0.66  QUALIFIED  59.04    27.69   
1     2     GBR          Adam PEATY  29  0.64  QUALIFIED  59.18    27.21   
2     3     AIN    Ilya SHYMANOVICH  29  0.63  QUALIFIED  59.25    27.35   
3     4     NED       Arno KAMMINGA  28  0.67  QUALIFIED  59.39    27.83   
4     4     ITA  Nicolo MARTINENGHI  24  0.65  QUALIFIED  59.39    27.66   

  100 Split                   Event  Round  
0     31.35  Mens 100M Breaststroke  Heats  
1     31.97  Mens 100M Breaststroke  Heats  
2     31.90  Mens 100M Breaststroke  Heats  
3     31.56  Mens 100M Breaststroke  Heats  
4     31.73  Mens 100M Breaststroke  Heats

Place Country               Name Age    RT   Status Time    50m   100m  \
0     1     GER      Lukas MARTENS  22  0.69  3:41.78       24.84  27.17   
1     2     AUS  Elijah WINNINGTON  24  0.67  3:42.21       25.21  27.57   
2     3     KOR         Woomin KIM  22  0.62  3:42.50       25.00  27.50   
3     4     AUS       Samuel SHORT  20  0.64  3:42.64       25.72  27.62   
4     5     BRA    Guilherme COSTA  25  0.70  3:42.76       25.57  28.04   

    150m   200m   250m   300m   350m   400m                Event  Round  
0  28.13  28.26  28.25  28.61  28.27  28.25  Mens 400M Freestyle  Heats  
1  28.03  28.37  28.70  28.56  28.44  27.33  Mens 400M Freestyle  Heats  
2  27.86  28.35  28.39  28.77  28.49  28.14  Mens 400M Freestyle  Heats  
3  27.80  28.27  28.30  28.51  28.59  27.83  Mens 400M Freestyle  Heats  
4  28.21  28.43  28.66  28.36  28.18  27.31  Mens 400M Freestyle  Heats

In [4]:
m100Breast_semis_df=parse_100_heats('M100Breast_semis_results.csv', 'Mens 100M Breastroke', 'Semmifinals')
m100Breast_semis_df=sort_heats(m100Breast_semis_df)

display(m100Breast_semis_df)

m100Breast_semis_df.to_csv(outdir / 'M100Breast_semis_results.csv', index=False)



Place Country                      Name Age    RT      Status  \
0       1     GBR                Adam PEATY  29  0.63   QUALIFIED   
1       2     CHN               Haiyang QIN  25  0.64   QUALIFIED   
2       3     NED             Arno KAMMINGA  28  0.68   QUALIFIED   
3       4     USA                  Nic FINK  31  0.68   QUALIFIED   
4       5     NED            Caspar CORBEAU  23  0.67   QUALIFIED   
5       6     ITA        Nicolo MARTINENGHI  24  0.65   QUALIFIED   
6       7     GER           Lucas MATZERATH  24  0.70   QUALIFIED   
7       8     GER             Melvin IMOUDU  25  0.68   QUALIFIED   
8       9     ITA  Ludovico Blu Art VIBERTI  22  0.66   QUALIFIED   
9      10     AIN          Ilya SHYMANOVICH  29  0.63   QUALIFIED   
10     11     GBR               James WILBY  30  0.69  ELIMINATED   
11     12     AUS               Joshua YONG  23  0.65  ELIMINATED   
12     13     AIN             Evgenii SOMOV  25  0.77  ELIMINATED   
13     14     USA           Charlie SWANSON  26  0.68  ELIMINATED   
14     15     AUT      Bernhard REITSHAMMER  30  0.68  ELIMINATED   
15     16     ISR              Ron POLONSKY  23  0.72  ELIMINATED   

                      Time 50 Split 100 Split                 Event  \
0  1900-01-01 00:00:58.860    26.90     31.96  Mens 100M Breastroke   
1  1900-01-01 00:00:58.930    27.12     31.81  Mens 100M Breastroke   
2  1900-01-01 00:00:59.120    27.64     31.48  Mens 100M Breastroke   
3  1900-01-01 00:00:59.160    27.44     31.72  Mens 100M Breastroke   
4  1900-01-01 00:00:59.240    27.70     31.54  Mens 100M Breastroke   
5  1900-01-01 00:00:59.280    27.43     31.85  Mens 100M Breastroke   
6  1900-01-01 00:00:59.310    27.72     31.59  Mens 100M Breastroke   
7  1900-01-01 00:00:59.380    27.67     31.71  Mens 100M Breastroke   
8  1900-01-01 00:00:59.380    28.18     31.20  Mens 100M Breastroke   
9  1900-01-01 00:00:59.450    27.16     32.29  Mens 100M Breastroke   
10 1900-01-01 00:00:59.490    27.91     31.58  Mens 100M Breastroke   
11 1900-01-01 00:00:59.640    28.37     31.27  Mens 100M Breastroke   
12 1900-01-01 00:01:00.000    27.66     32.34  Mens 100M Breastroke   
13 1900-01-01 00:01:00.160    28.14     32.02  Mens 100M Breastroke   
14 1900-01-01 00:01:00.180    27.79     32.39  Mens 100M Breastroke   
15 1900-01-01 00:01:00.370    28.23     32.14  Mens 100M Breastroke   

          Round  
0   Semmifinals  
1   Semmifinals  
2   Semmifinals  
3   Semmifinals  
4   Semmifinals  
5   Semmifinals  
6   Semmifinals  
7   Semmifinals  
8   Semmifinals  
9   Semmifinals  
10  Semmifinals  
11  Semmifinals  
12  Semmifinals  
13  Semmifinals  
14  Semmifinals  
15  Semmifinals

In [8]:
#parse mens 400 free finals

m400Fr_finals_df=parse_400_heats('M400Free_finals_results.csv', 'Mens 400M Freestyle', 'Finals')

m400Fr_finals_df.to_csv(outdir / 'M400Free_finals_results.csv', index=False)

display(m400Fr_finals_df)


w400Fr_finals_df=parse_400_heats('W400Free_finals_results.csv', 'Womens 400M Freestyle', 'Finals')
w400Fr_finals_df.to_csv(outdir / 'W400Free_finals_results.csv', index=False)
display(w400Fr_finals_df)

Place Country               Name Age    RT   Status Time    50m   100m  \
0     1     GER      Lukas MARTENS  22  0.69  3:41.78       24.84  27.17   
1     2     AUS  Elijah WINNINGTON  24  0.67  3:42.21       25.21  27.57   
2     3     KOR         Woomin KIM  22  0.62  3:42.50       25.00  27.50   
3     4     AUS       Samuel SHORT  20  0.64  3:42.64       25.72  27.62   
4     5     BRA    Guilherme COSTA  25  0.70  3:42.76       25.57  28.04   
5     6     CHN          Liwei FEI  21  0.66  3:44.24       25.62  27.61   
6     7     GER      Oliver KLEMET  22  0.68  3:46.59       26.31  28.01   
7     8     USA     Aaron SHACKELL  19  0.67  3:47.00       25.73  28.08   

    150m   200m   250m   300m   350m   400m                Event   Round  
0  28.13  28.26  28.25  28.61  28.27  28.25  Mens 400M Freestyle  Finals  
1  28.03  28.37  28.70  28.56  28.44  27.33  Mens 400M Freestyle  Finals  
2  27.86  28.35  28.39  28.77  28.49  28.14  Mens 400M Freestyle  Finals  
3  27.80  28.27  28.30  28.51  28.59  27.83  Mens 400M Freestyle  Finals  
4  28.21  28.43  28.66  28.36  28.18  27.31  Mens 400M Freestyle  Finals  
5  28.46  28.60  28.88  28.79  28.37  27.91  Mens 400M Freestyle  Finals  
6  28.33  28.57  28.69  29.02  28.97  28.69  Mens 400M Freestyle  Finals  
7  28.92  28.93  28.92  29.13  29.20  28.09  Mens 400M Freestyle  Finals

Place Country                                       Name Age    RT   Status  \
0     1     AUS                             Ariarne TITMUS  23  0.72  3:57.49   
1     2     CAN                            Summer MCINTOSH  17  0.74  3:58.37   
2     3     USA                              Katie LEDECKY  27  0.68  4:00.86   
3     4     NZL                          Erika FAIRWEATHER  20  0.75  4:01.12   
4     5     GER                                Isabel GOSE  22  0.75  4:02.14   
5     6     USA                               Paige MADDEN  25  0.73  4:02.26   
6     7     BRA  Maria Fernanda DE OLIVEIRA DA SILVA COSTA  21  0.72  4:03.53   
7     8     AUS                          Jamie Ann PERKINS  19  0.76  4:04.96   

  Time    50m   100m   150m   200m   250m   300m   350m   400m  \
0       27.31  29.61  30.04  30.01  29.85  30.10  30.44  30.13   
1       27.41  29.62  30.33  29.96  30.16  30.26  30.46  30.17   
2       27.79  29.90  30.20  30.63  30.13  30.86  30.59  30.76   
3       27.71  29.94  30.59  30.74  30.90  31.28  30.46  29.50   
4       28.20  29.97  30.40  30.62  30.61  30.95  31.08  30.31   
5       27.88  29.89  30.04  30.68  30.79  30.83  31.31  30.84   
6       28.03  30.22  30.72  31.30  31.27  31.36  30.74  29.89   
7       27.86  30.33  30.61  31.12  31.17  31.47  31.26  31.14   

                   Event   Round  
0  Womens 400M Freestyle  Finals  
1  Womens 400M Freestyle  Finals  
2  Womens 400M Freestyle  Finals  
3  Womens 400M Freestyle  Finals  
4  Womens 400M Freestyle  Finals  
5  Womens 400M Freestyle  Finals  
6  Womens 400M Freestyle  Finals  
7  Womens 400M Freestyle  Finals

In [3]:
#parse the womens 100 fly semis
from utils import parse_100_heats

w100Fly_semis_df=parse_100_heats('W100Fly_semis_results.csv', 'Womens 100M Butterfly', 'Semifinals')

w100Fly_semis_df=sort_heats(w100Fly_semis_df)



display(w100Fly_semis_df)



Place Country                    Name Age    RT     Status  \
0       1     USA          Gretchen WALSH  21  0.76  QUALIFIED   
1       2     USA             Torri HUSKE  21  0.61  QUALIFIED   
2       3     CHN             Yufei ZHANG  26  0.63  QUALIFIED   
3       4     GER         Angelina KOHLER  23  0.74  QUALIFIED   
4       5     CAN        Margaret MACNEIL  24  0.67  QUALIFIED   
5       6     AUS             Emma MCKEON  30  0.71  QUALIFIED   
6       7     JPN            Mizuki HIRAI  17  0.59  QUALIFIED   
7       8     SWE          Louise HANSSON  27  0.72  QUALIFIED   
8       9     FRA            Marie WATTEL  27  0.70  QUALIFIED   
9      10     BEL       Roos VANOTTERDIJK  19  0.66  QUALIFIED   
10     11     CZE       Barbora SEEMANOVA  24  0.62  QUALIFIED   
11     12     JPN             Rikako IKEE  24  0.67  QUALIFIED   
12     13     AUS      Alexandria PERKINS  24  0.69  QUALIFIED   
13     14     RSA          Erin GALLAGHER  25  0.70  QUALIFIED   
14     15     NED             Tessa GIELE  21  0.61  QUALIFIED   
15     16     GBR  Keanna Louise MACINNES  22  0.66  QUALIFIED   

                      Time 50 Split 100 Split                  Event  \
0  1900-01-01 00:00:55.380    25.29     30.09  Womens 100M Butterfly   
1  1900-01-01 00:00:56.000    25.69     30.31  Womens 100M Butterfly   
2  1900-01-01 00:00:56.150    25.81     30.34  Womens 100M Butterfly   
3  1900-01-01 00:00:56.550    26.38     30.17  Womens 100M Butterfly   
4  1900-01-01 00:00:56.550    26.34     30.21  Womens 100M Butterfly   
5  1900-01-01 00:00:56.740    26.18     30.56  Womens 100M Butterfly   
6  1900-01-01 00:00:56.800    26.12     30.68  Womens 100M Butterfly   
7  1900-01-01 00:00:56.930    26.51     30.42  Womens 100M Butterfly   
8  1900-01-01 00:00:57.240    26.58     30.66  Womens 100M Butterfly   
9  1900-01-01 00:00:57.250    26.48     30.77  Womens 100M Butterfly   
10 1900-01-01 00:00:57.640    26.35     31.29  Womens 100M Butterfly   
11 1900-01-01 00:00:57.790    26.36     31.43  Womens 100M Butterfly   
12 1900-01-01 00:00:57.840    26.50     31.34  Womens 100M Butterfly   
13 1900-01-01 00:00:57.900    26.62     31.28  Womens 100M Butterfly   
14 1900-01-01 00:00:57.910    26.43     31.48  Womens 100M Butterfly   
15 1900-01-01 00:00:58.110    27.18     30.93  Womens 100M Butterfly   

         Round  
0   Semifinals  
1   Semifinals  
2   Semifinals  
3   Semifinals  
4   Semifinals  
5   Semifinals  
6   Semifinals  
7   Semifinals  
8   Semifinals  
9   Semifinals  
10  Semifinals  
11  Semifinals  
12  Semifinals  
13  Semifinals  
14  Semifinals  
15  Semifinals

## Day 2

M200 free semis,

M400IM heats

W100 breast heats

M100 back heats

W200 free heats

M400 IM finals

W100 fly finals

M200 free semis

W100 breast semis

M100 back semis

M100 breast finals

W200 free semis




There is a new King of France, and his name is Leon Marchand! The defending world champion wins his first Olympic gold medal in the 400 IM in front of the home crowd!

Americans Torri Huske and Gretchen Walsh go 1-2 in Women's 100 Fly! I was thinking that Walsh would get the gold, but its Huske who gets the win. The margin of victory for Huske was .04 seconds. Interestingly, Walsh's reaction time off the blocks was slowest in the field at .76, while Huske's was second fastest at .63.  Walsh's 50 split was the fastest in the field at 25.40. Had she had a faster reaction time, she would have won the gold. But this is a great win for Huske, who was .01 off the medal stand in the same event in Tokyo, finshing 4th. 

Nicolo Martinenghi wins his first Olympic gold in the Mens 100 breastroke.  A three-Peaty would not happen, as Adam Peaty finishes tied for Silver with Nick Fink. Peaty was the 2-time defending Olympic champion in the event. He did not particapte in World Championships last year, taking time away from the sport for mental health reasons. He had a rather pedestrian second 50 split at 31.97. The whole field was a bit slower than I expected. Peaty's time in semifinals yesterday would have won the gold today. 

Nick Fink, whose day job is an engineer, won his first Olympic medal. He was the oldest swimmer in the final at 31. Fink has even bigger things on the horizon as his wife, former Olympian Melanie Margalis, is expecting their first child in September.

In [4]:
from utils import parse_200_heats
from utils import sort_heats
m200Free_heats_df=parse_200_heats('M200Free_heats_results.csv', 'Mens 200M Freestyle', 'Heats')
m200Free_heats_df=sort_heats(m200Free_heats_df)
m200Free_heats_df.to_csv(outdir / 'M200Free_heats_results.csv', index=False)

m100Back_heats_df=parse_100_heats('M100Back_heats_results.csv', 'Mens 100M Backstroke', 'Heats')
m100Back_heats_df=sort_heats(m100Back_heats_df)
m100Back_heats_df.to_csv(outdir / 'M100Back_heats_results.csv', index=False)


w100Breast_heats_df=parse_100_heats('W100Breast_heats_results.csv', 'Womens 100M Breaststroke', 'Heats')
w100Breast_heats_df=sort_heats(w100Breast_heats_df)
w100Breast_heats_df.to_csv(outdir / 'W100Breast_heats_results.csv', index=False)

m400IM_heats_df=parse_400_heats('M400IM_heats_results.csv', 'Mens 400M Individual Medley', 'Heats')
m400IM_heats_df=sort_heats(m400IM_heats_df)
m400IM_heats_df.to_csv(outdir / 'M400IM_heats_results.csv', index=False)


w200Free_heats_df=parse_200_heats('W200Free_heats_results.csv', 'Womens 200M Freestyle', 'Heats')
w200Free_heats_df=sort_heats(w200Free_heats_df)
w200Free_heats_df.to_csv(outdir / 'W200Free_heats_results.csv', index=False)


w100Fly_finals_df=parse_100_heats('W100Fly_finals_results.csv', 'Womens 100M Butterfly', 'Finals')
w100Fly_finals_df.to_csv(outdir / 'W100Fly_finals_results.csv', index=False)


m200Free_semis_df=parse_200_heats('M200Free_semis_results.csv', 'Mens 200M Freestyle', 'Semifinals')
m200Free_semis_df=sort_heats(m200Free_semis_df)
m200Free_semis_df.to_csv(outdir / 'M200Free_semis_results.csv', index=False)

w100Breast_semis_df=parse_100_heats('W100Breast_semis_results.csv', 'Womens 100M Breaststroke', 'Semifinals')
w100Breast_semis_df=sort_heats(w100Breast_semis_df)
w100Breast_semis_df.to_csv(outdir / 'W100Breast_semis_results.csv', index=False)

m100Back_semis_df=parse_100_heats('M100Back_semis_results.csv', 'Mens 100M Backstroke', 'Semifinals')
m100Back_semis_df=sort_heats(m100Back_semis_df)
m100Back_semis_df.to_csv(outdir / 'M100Back_semis_results.csv', index=False)

m100Breast_finals_df=parse_100_heats('M100Breast_finals_results.csv', 'Mens 100M Breaststroke', 'Finals')
m100Breast_finals_df.to_csv(outdir / 'M100Breast_finals_results.csv', index=False)

m400IM_finals_df=parse_400_heats('M400IM_finals_results.csv', 'Mens 400M Individual Medley', 'Finals')
m400IM_finals_df.to_csv(outdir / 'M400IM_finals_results.csv', index=False)




display(m200Free_heats_df)
display(m400IM_heats_df)
display(m100Back_heats_df)
display(w200Free_heats_df)
display(w100Fly_finals_df)
display(m200Free_semis_df)
display(w100Breast_semis_df)
display(m100Back_semis_df)
display(m100Breast_finals_df)
display(m400IM_finals_df)


Place Country                     Name Age    RT      Status  \
0       1     ROU           David POPOVICI  19  0.67   QUALIFIED   
1       2     LTU             Danas RAPSYS  29  0.67   QUALIFIED   
2       3     BEL  Lucas Pierre A HENVEAUX  23  0.66   QUALIFIED   
3       4     KOR             Sunwoo HWANG  21  0.63   QUALIFIED   
4       5     AUS     Maximillian GIULIANI  21  0.67   QUALIFIED   
5       6     GBR         Matthew RICHARDS  21  0.61   QUALIFIED   
6       7     USA              Luke HOBSON  21  0.78   QUALIFIED   
7       8     JPN      Katsuhiro MATSUMOTO  27  0.69   QUALIFIED   
8       9     AUS             Thomas NEILL  22  0.70   QUALIFIED   
9      10     GER            Lukas MARTENS  22  0.71   QUALIFIED   
10     11     GBR             Duncan SCOTT  27  0.65   QUALIFIED   
11     12     KOR               Woomin KIM  22  0.60   QUALIFIED   
12     13     GER          Rafael MIROSLAW  23  0.65   QUALIFIED   
13     14     ISR             Denis LOKTEV  24  0.63   QUALIFIED   
14     15     ITA       Alessandro RAGAINI  17  0.70   QUALIFIED   
15     16     ITA            Filippo MEGLI  27  0.64   QUALIFIED   
16     17     SUI         Antonio DJAKOVIC  21  0.67   QUALIFIED   
17     18     SRB      Velimir STJEPANOVIC  30  0.65  ELIMINATED   
18     19     USA           Chris GUILIANO  21  0.62  ELIMINATED   
19     20     MEX                Jorge IGA  27  0.65  ELIMINATED   
20     21     SLO              Saso BOSKAN  21  0.63  ELIMINATED   
21     22     CHN               Zhanle PAN  19  0.65  ELIMINATED   
22     23     CHN                Xinjie JI  26  0.71  ELIMINATED   
23     24     SYR              Omar ABBASS  25  0.67  ELIMINATED   
24     25     PAK   Muhammad Ahmed DURRANI  18  0.68  ELIMINATED   

                      Time    50m   100m   150m   200m                Event  \
0  1900-01-01 00:01:45.650  24.17  26.98  27.25  27.25  Mens 200M Freestyle   
1  1900-01-01 00:01:45.910  24.59  26.82  27.69  26.81  Mens 200M Freestyle   
2  1900-01-01 00:01:46.040  25.16  26.90  27.43  26.55  Mens 200M Freestyle   
3  1900-01-01 00:01:46.130  24.46  27.15  27.43  27.09  Mens 200M Freestyle   
4  1900-01-01 00:01:46.150  24.91  27.18  27.07  26.99  Mens 200M Freestyle   
5  1900-01-01 00:01:46.190  24.83  27.12  27.44  26.80  Mens 200M Freestyle   
6  1900-01-01 00:01:46.230  24.89  27.09  27.11  27.14  Mens 200M Freestyle   
7  1900-01-01 00:01:46.230  24.76  26.86  27.82  26.79  Mens 200M Freestyle   
8  1900-01-01 00:01:46.270  25.25  27.11  27.31  26.60  Mens 200M Freestyle   
9  1900-01-01 00:01:46.330  24.40  26.79  27.75  27.39  Mens 200M Freestyle   
10 1900-01-01 00:01:46.340  24.78  26.93  27.48  27.15  Mens 200M Freestyle   
11 1900-01-01 00:01:46.640  24.82  26.88  27.63  27.31  Mens 200M Freestyle   
12 1900-01-01 00:01:46.810  24.71  26.94  27.35  27.81  Mens 200M Freestyle   
13 1900-01-01 00:01:47.010  24.96  27.14  27.19  27.72  Mens 200M Freestyle   
14 1900-01-01 00:01:47.310  25.33  26.73  27.62  27.63  Mens 200M Freestyle   
15 1900-01-01 00:01:47.390  24.71  27.20  27.85  27.63  Mens 200M Freestyle   
16 1900-01-01 00:01:47.460  24.28  27.23  28.10  27.85  Mens 200M Freestyle   
17 1900-01-01 00:01:47.560  24.81  27.23  27.85  27.67  Mens 200M Freestyle   
18 1900-01-01 00:01:47.600  24.48  27.57  27.90  27.65  Mens 200M Freestyle   
19 1900-01-01 00:01:48.380  24.92  27.02  27.43  29.01  Mens 200M Freestyle   
20 1900-01-01 00:01:48.750  25.06  27.37  27.95  28.37  Mens 200M Freestyle   
21 1900-01-01 00:01:49.470  25.12  27.57  27.90  28.88  Mens 200M Freestyle   
22 1900-01-01 00:01:49.880  25.18  27.44  28.43  28.83  Mens 200M Freestyle   
23 1900-01-01 00:01:53.010  26.37  28.16  29.22  29.26  Mens 200M Freestyle   
24 1900-01-01 00:01:58.670  26.62  29.21  31.06  31.78  Mens 200M Freestyle   

    Round  
0   Heats  
1   Heats  
2   Heats  
3   Heats  
4   Heats  
5   Heats  
6   Heats  
7   Heats  
8   Heats  
9   Heats  
10  Heats  
11  Heats  
12  Heats  
13  Heats  


Place Country                     Name Age    RT     Status  \
0       1     FRA            Leon MARCHAND  22  0.67  QUALIFIED   
1       2     GBR           Max LITCHFIELD  29  0.62  QUALIFIED   
2       3     JPN               Daiya SETO  30  0.59  QUALIFIED   
3       4     USA            Carson FOSTER  22  0.67  QUALIFIED   
4       5     JPN      Tomoyuki MATSUSHITA  18  0.64  QUALIFIED   
5       6     ITA         Alberto RAZZETTI  25  0.64  QUALIFIED   
6       7     NZL          Lewis CLAREBURT  25  0.69  QUALIFIED   
7       8     GER           Cedric BUSSING  20  0.69  QUALIFIED   
8       9     HUN             Balazs HOLLO  25  0.66  QUALIFIED   
9      10     CHN           Zhanshuo ZHANG  17  0.69  QUALIFIED   
10     11     USA             Chase KALISZ  30  0.74  QUALIFIED   
11     12     AUS  William Campbell PETRIC  19  0.71  QUALIFIED   
12     13     AUS            Brendon SMITH  24  0.70  QUALIFIED   
13     14     HUN            Gabor ZOMBORI  21  0.73  QUALIFIED   
14     15     GRE     Apostolos PAPASTAMOS  23  0.66  QUALIFIED   
15     16     CAN        Tristan JANKOVICS  20  0.67  QUALIFIED   

                      Time    50m   100m   150m   200m   250m   300m   350m  \
0  1900-01-01 00:04:08.300  25.44  29.08  31.96  30.83  34.69  35.92  31.06   
1  1900-01-01 00:04:09.510  26.18  29.95  31.69  31.37  35.59  36.12  29.49   
2  1900-01-01 00:04:10.920  25.41  29.08  32.19  31.77  35.64  36.17  30.24   
3  1900-01-01 00:04:11.070  25.96  29.50  32.52  31.70  35.64  36.25  30.33   
4  1900-01-01 00:04:11.180  25.95  30.38  32.99  32.02  35.56  36.24  29.93   
5  1900-01-01 00:04:11.520  25.98  29.81  32.67  32.10  35.15  36.17  30.11   
6  1900-01-01 00:04:11.520  26.25  29.87  32.81  31.11  36.39  36.34  30.31   
7  1900-01-01 00:04:11.520  26.82  30.85  32.83  31.37  35.26  35.57  30.39   
8  1900-01-01 00:04:12.200  26.46  29.92  33.23  32.37  35.86  35.75  30.09   
9  1900-01-01 00:04:12.710  26.42  30.93  33.15  31.79  35.87  36.81  30.17   
10 1900-01-01 00:04:13.360  26.34  30.04  32.94  32.21  35.21  36.06  30.92   
11 1900-01-01 00:04:13.580  26.29  30.40  33.16  32.53  35.07  35.99  30.49   
12 1900-01-01 00:04:14.360  26.27  30.15  32.52  32.08  36.36  37.65  30.16   
13 1900-01-01 00:04:14.880  26.10  30.70  33.03  32.76  36.10  36.34  30.72   
14 1900-01-01 00:04:15.320  26.29  30.27  32.27  32.28  35.34  36.90  30.64   
15 1900-01-01 00:04:18.230  26.79  30.76  32.92  32.84  36.39  37.55  30.69   

     400m                        Event  Round  
0   29.32  Mens 400M Individual Medley  Heats  
1   29.12  Mens 400M Individual Medley  Heats  
2   30.42  Mens 400M Individual Medley  Heats  
3   29.17  Mens 400M Individual Medley  Heats  
4   28.11  Mens 400M Individual Medley  Heats  
5   29.53  Mens 400M Individual Medley  Heats  
6   28.44  Mens 400M Individual Medley  Heats  
7   28.43  Mens 400M Individual Medley  Heats  
8   28.52  Mens 400M Individual Medley  Heats  
9   27.57  Mens 400M Individual Medley  Heats  
10  29.64  Mens 400M Individual Medley  Heats  
11  29.65  Mens 400M Individual Medley  Heats  
12  29.17  Mens 400M Individual Medley  Heats  
13  29.13  Mens 400M Individual Medley  Heats  
14  31.33  Mens 400M Individual Medley  Heats  
15  30.29  Mens 400M Individual Medley  Heats

Place Country                             Name Age    RT      Status  \
0       1     HUN                       Hubert KOS  21  0.52   QUALIFIED   
1       2     RSA                    Pieter COETZE  20  0.68   QUALIFIED   
2       3     POL                   Ksawery MASIUK  19  0.63   QUALIFIED   
3       4     CHN                         Jiayu XU  28  0.56   QUALIFIED   
4       5     FRA             Yohann NDOYE-BROUARD  23  0.55   QUALIFIED   
5       6     GRE           Evangelos MAKRYGIANNIS  24  0.59   QUALIFIED   
6       7     USA                 Hunter ARMSTRONG  23  0.61   QUALIFIED   
7       8     CZE                  Miroslav KNEDLA  19  0.53   QUALIFIED   
8       9     ITA                    Thomas CECCON  23  0.63   QUALIFIED   
9      10     ESP                    Hugo GONZALEZ  25  0.59   QUALIFIED   
10     11     GER                 Ole BRAUNSCHWEIG  26  0.61   QUALIFIED   
11     12     HUN                       Adam JASZO  22  0.64   QUALIFIED   
12     13     CAN                   Javier ACEVEDO  26  0.59   QUALIFIED   
13     14     NED                Kai VAN WESTERING  20  0.69   QUALIFIED   
14     15     AUS                     Isaac COOPER  20  0.54   QUALIFIED   
15     16     ITA                 Michele LAMBERTI  23  0.55   QUALIFIED   
16     17     UKR             Oleksandr ZHELTIAKOV  18  0.56   QUALIFIED   
17     18     SUI                   Thierry BOLLIN  24  0.60  ELIMINATED   
18     19     ISV               Maximillian WILSON  20  0.55  ELIMINATED   
19     20     ISR                     Adam MARAANA  21  0.54  ELIMINATED   
20     21     GER                     Marek ULRICH  27  0.66  ELIMINATED   
21     22     KOR                         Juho LEE  29  0.60  ELIMINATED   
22     23     JPN                   Riku MATSUYAMA  22  0.51  ELIMINATED   
23     24     POR                       Joao COSTA  22  0.58  ELIMINATED   
24     25     NZL                     Kane FOLLOWS  27  0.63  ELIMINATED   
25     26     IND                  Srihari NATARAJ  23  0.58  ELIMINATED   
26     27     ARG                   Ulises SARAVIA  18  0.53  ELIMINATED   
27     28     AUT             Bernhard REITSHAMMER  30  0.64  ELIMINATED   
28     29     COL           Anthony RINCON VELASCO  23  0.52  ELIMINATED   
29     30     PUR                   Yeziel MORALES  28  0.52  ELIMINATED   
30     31     BER                      Jack HARVEY  21  0.56  ELIMINATED   
31     32     TUR                       Berke SAKA  21  0.58  ELIMINATED   
32     33     GEO                   Noe PANTSKHAVA  18  0.57  ELIMINATED   
33     34     BEN  Alexis Michel Roger Dodji KPADE  18  0.59  ELIMINATED   
34     35     PLE                  Yazan AL BAWWAB  24  0.61  ELIMINATED   
35     36     GRN                  Zackary GRESHAM  19  0.58  ELIMINATED   
36     37     BRU                        Zeke CHAN  16  0.58  ELIMINATED   
37     38     TGA             Alan Koti Lopeti UHI  18  0.69  ELIMINATED   

                      Time 50 Split 100 Split                 Event  Round  
0  1900-01-01 00:00:52.780    25.71     27.07  Mens 100M Backstroke  Heats  
1  1900-01-01 00:00:52.900    25.49     27.41  Mens 100M Backstroke  Heats  
2  1900-01-01 00:00:53.080    25.14     27.94  Mens 100M Backstroke  Heats  
3  1900-01-01 00:00:53.200    25.75     27.45  Mens 100M Backstroke  Heats  
4  1900-01-01 00:00:53.200    25.99     27.21  Mens 100M Backstroke  Heats  
5  1900-01-01 00:00:53.240    25.67     27.57  Mens 100M Backstroke  Heats  
6  1900-01-01 00:00:53.340    25.64     27.70  Mens 100M Backstroke  Heats  
7  1900-01-01 00:00:53.410    25.34     28.07  Mens 100M Backstroke  Heats  
8  1900-01-01 00:00:53.450    25.71     27.74  Mens 100M Backstroke  Heats  
9  1900-01-01 00:00:53.680    26.25     27.43  Mens 100M Backstroke  Heats  
10 1900-01-01 00:00:53.950    25.86     28.09  Mens 100M Backstroke  Heats  
11 1900-01-01 00:00:53.970    26.11     27.86  Mens 100M Backstroke  Heats  
12 1900-01-01 00:00:54.190    26.31   

Place Country                                       Name Age    RT  \
0       1     AUS                         Mollie O'CALLAGHAN  20  0.71   
1       2     CAN                         Mary-Sophie HARVEY  24  0.72   
2       3     AUS                             Ariarne TITMUS  23  0.76   
3       4     CHN                                 Bingjie LI  22  0.67   
4       5     HKG                 Siobhan Bernadette HAUGHEY  26  0.70   
5       6     USA                           Claire WEINSTEIN  17  0.75   
6       7     NZL                          Erika FAIRWEATHER  20  0.76   
7       8     BRA  Maria Fernanda DE OLIVEIRA DA SILVA COSTA  21  0.72   
8       9     CHN                               Junxuan YANG  22  0.68   
9      10     CZE                          Barbora SEEMANOVA  24  0.64   
10     11     USA                               Erin GEMMELL  19  0.76   
11     12     BEL                           Valentine DUMONT  24  0.69   
12     13     HUN                        Lilla Minna ABRAHAM  18  0.70   
13     14     RSA                                Aimee CANNY  20  0.75   
14     15     ISL              Snaefridur Sol JORUNNARDOTTIR  23  0.73   
15     16     ROU                   Rebecca-Aimee DIACONESCU  18  0.74   
16     17     GER                            Julia MROZINSKI  24  0.70   
17     18     MGL                       Enkhkhuslen BATBAYAR  22  0.71   
18     19     ISR                               Lea POLONSKY  22  0.66   
19     20     VEN               Maria Victoria YEGRES COTTIN  18  0.70   
20     21     CUB                              Andrea BECALI  20  0.73   
21     22     HON                       Julimar AVILA MANCIA  27  0.66   
22     23     IND                          Dhinidhi DESINGHU  14  0.70   
23     24     EGY                       Lojine ABDALLA SALAH  19  0.72   
24     25     LAO                   Ariana Southa DIRKZWAGER  20  0.70   
25     26     PAK                              Jehanara NABI  20  0.73   
26     27     ALB                                Kaltra MECA  16  0.76   
27     28     UAE                              Maha ALSHEHHI  18  0.66   
28     29     KSA                   Mashael Meshari A ALAYED  17  0.75   
29     30     NEP                                 Duana LAMA  18  0.67   

        Status                    Time    50m   100m   150m   200m  \
0    QUALIFIED 1900-01-01 00:01:55.790  27.48  29.74  30.06  28.51   
1    QUALIFIED 1900-01-01 00:01:56.210  27.85  30.01  29.05  29.30   
2    QUALIFIED 1900-01-01 00:01:56.230  27.19  29.39  30.14  29.51   
3    QUALIFIED 1900-01-01 00:01:56.280  27.67  29.44  29.83  29.34   
4    QUALIFIED 1900-01-01 00:01:56.380  27.02  29.57  29.98  29.81   
5    QUALIFIED 1900-01-01 00:01:56.480  27.49  29.41  29.95  29.63   
6    QUALIFIED 1900-01-01 00:01:56.540  27.57  29.52  30.05  29.40   
7    QUALIFIED 1900-01-01 00:01:56.650  27.62  29.44  30.00  29.59   
8    QUALIFIED 1900-01-01 00:01:56.830  26.99  29.92  30.30  29.62   
9    QUALIFIED 1900-01-01 00:01:57.020  27.17  29.89  30.31  29.65   
10   QUALIFIED 1900-01-01 00:01:57.230  27.16  29.61  30.22  30.24   
11   QUALIFIED 1900-01-01 00:01:57.500  27.74  29.59  30.14  30.03   
12   QUALIFIED 1900-01-01 00:01:57.770  28.04  30.01  29.98  29.74   
13   QUALIFIED 1900-01-01 00:01:57.810  27.52  29.78  30.17  30.34   
14   QUALIFIED 1900-01-01 00:01:58.320  27.96  29.85  29.70  30.81   
15   QUALIFIED 1900-01-01 00:01:59.290  27.82  30.36  31.01  30.10   
16   QUALIFIED 1900-01-01 00:01:59.870  27.82  30.22  31.01  30.82   
17  ELIMINATED 1900-01-01 00:01:59.940  28.14  30.32  30.75  30.73   
18  ELIMINATED 1900-01-01 00:02:00.380  28.42  30.15  30.63  31.18   
19  ELIMINATED 1900-01-01 00:02:00.660  28.30  30.02  30.83  31.51   
20  ELIMINATED 1900-01-01 00:02:03.380  28.45  30.65  31.94  32.34   
21  ELIMINATED 1900-01-01 00:02:04.880  28.51  31.54  32.67  32.16   
22  ELIMINATED 1900-01-01 00:02:06.960  29.68  32.07  32.71  32.50   
23  ELIMINATED 1900-01-01 00:

Place Country              Name Age    RT Status Time 50 Split 100 Split  \
0     1     USA       Torri HUSKE  21  0.63  55.59         25.61     29.98   
1     2     USA    Gretchen WALSH  21  0.75  55.63         25.40     30.23   
2     3     CHN       Yufei ZHANG  26  0.66  56.21         25.42     30.79   
3     4     GER   Angelina KOHLER  23  0.72  56.42         26.35     30.07   
4     5     CAN  Margaret MACNEIL  24  0.68  56.44         25.94     30.50   
5     6     AUS       Emma MCKEON  30  0.71  56.93         26.40     30.53   
6     7     JPN      Mizuki HIRAI  17  0.59  57.19         26.66     30.53   
7     8     SWE    Louise HANSSON  27  0.71  57.34         26.61     30.73   

                   Event   Round  
0  Womens 100M Butterfly  Finals  
1  Womens 100M Butterfly  Finals  
2  Womens 100M Butterfly  Finals  
3  Womens 100M Butterfly  Finals  
4  Womens 100M Butterfly  Finals  
5  Womens 100M Butterfly  Finals  
6  Womens 100M Butterfly  Finals  
7  Womens 100M Butterfly  Finals

Place Country                     Name Age    RT     Status  \
0       1     ROU           David POPOVICI  19  0.67  QUALIFIED   
1       2     GBR             Duncan SCOTT  27  0.65  QUALIFIED   
2       3     USA              Luke HOBSON  21  0.76  QUALIFIED   
3       4     GER            Lukas MARTENS  22  0.69  QUALIFIED   
4       5     AUS     Maximillian GIULIANI  21  0.65  QUALIFIED   
5       6     LTU             Danas RAPSYS  29  0.67  QUALIFIED   
6       7     GBR         Matthew RICHARDS  21  0.61  QUALIFIED   
7       8     JPN      Katsuhiro MATSUMOTO  27  0.68  QUALIFIED   
8       9     KOR             Sunwoo HWANG  21  0.59  QUALIFIED   
9      10     AUS             Thomas NEILL  22  0.70  QUALIFIED   
10     11     BEL  Lucas Pierre A HENVEAUX  23  0.65  QUALIFIED   
11     12     KOR               Woomin KIM  22  0.60  QUALIFIED   
12     13     ITA            Filippo MEGLI  27  0.65  QUALIFIED   
13     14     ITA       Alessandro RAGAINI  17  0.68  QUALIFIED   
14     15     GER          Rafael MIROSLAW  23  0.66  QUALIFIED   
15     16     ISR             Denis LOKTEV  24  0.64  QUALIFIED   

                      Time    50m   100m   150m   200m                Event  \
0  1900-01-01 00:01:44.530  24.12  26.76  26.75  26.90  Mens 200M Freestyle   
1  1900-01-01 00:01:44.940  24.77  26.49  27.01  26.67  Mens 200M Freestyle   
2  1900-01-01 00:01:45.190  24.78  26.66  27.23  26.52  Mens 200M Freestyle   
3  1900-01-01 00:01:45.360  24.42  26.73  27.35  26.86  Mens 200M Freestyle   
4  1900-01-01 00:01:45.370  24.69  26.82  26.85  27.01  Mens 200M Freestyle   
5  1900-01-01 00:01:45.480  24.50  26.78  27.05  27.15  Mens 200M Freestyle   
6  1900-01-01 00:01:45.630  24.20  26.90  27.40  27.13  Mens 200M Freestyle   
7  1900-01-01 00:01:45.880  24.61  26.69  27.53  27.05  Mens 200M Freestyle   
8  1900-01-01 00:01:45.920  24.10  26.85  27.67  27.30  Mens 200M Freestyle   
9  1900-01-01 00:01:46.180  25.05  26.84  27.39  26.90  Mens 200M Freestyle   
10 1900-01-01 00:01:46.200  24.75  26.89  27.25  27.31  Mens 200M Freestyle   
11 1900-01-01 00:01:46.580  24.68  26.75  27.62  27.53  Mens 200M Freestyle   
12 1900-01-01 00:01:46.870  24.68  26.89  27.66  27.64  Mens 200M Freestyle   
13 1900-01-01 00:01:47.080  25.33  26.84  27.64  27.27  Mens 200M Freestyle   
14 1900-01-01 00:01:47.340  24.60  27.01  27.84  27.89  Mens 200M Freestyle   
15 1900-01-01 00:01:47.930  24.72  26.94  27.93  28.34  Mens 200M Freestyle   

    Round  
0   Heats  
1   Heats  
2   Heats  
3   Heats  
4   Heats  
5   Heats  
6   Heats  
7   Heats  
8   Heats  
9   Heats  
10  Heats  
11  Heats  
12  Heats  
13  Heats  
14  Heats  
15  Heats

Place Country                 Name Age    RT     Status  \
0       1     RSA  Tatjana SCHOENMAKER  27  0.67  QUALIFIED   
1       2     IRL       Mona MC SHARRY  23  0.67  QUALIFIED   
2       3     USA           Lilly KING  27  0.67  QUALIFIED   
3       4     CHN        Qianting TANG  20  0.71  QUALIFIED   
4       5     AIN        Alina ZMUSHKA  27  0.70  QUALIFIED   
5       6     GBR       Angharad EVANS  21  0.67  QUALIFIED   
6       7     ITA     Benedetta PILATO  19  0.66  QUALIFIED   
7       8     EST       Eneli JEFIMOVA  17  0.66  QUALIFIED   
8       9     ITA       Lisa ANGIOLINI  29  0.70  QUALIFIED   
9      10     NED         Tes SCHOUTEN  23  0.66  QUALIFIED   
10     11     LTU       Ruta MEILUTYTE  27  0.58  QUALIFIED   
11     12     JPN        Satomi SUZUKI  33  0.63  QUALIFIED   
12     13     SWE       Sophie HANSSON  25  0.71  QUALIFIED   
13     14     CHN           Chang YANG  22  0.68  QUALIFIED   
14     15     ARG    Macarena CEBALLOS  29  0.68  QUALIFIED   
15     16     LTU  Kotryna TETEREVKOVA  22  0.68  QUALIFIED   

                      Time 50 Split 100 Split                     Event  \
0  1900-01-01 00:01:05.000    30.63     34.37  Womens 100M Breaststroke   
1  1900-01-01 00:01:05.510    30.96     34.55  Womens 100M Breaststroke   
2  1900-01-01 00:01:05.640    30.96     34.68  Womens 100M Breaststroke   
3  1900-01-01 00:01:05.830    30.60     35.23  Womens 100M Breaststroke   
4  1900-01-01 00:01:05.930    30.89     35.04  Womens 100M Breaststroke   
5  1900-01-01 00:01:05.990    31.43     34.56  Womens 100M Breaststroke   
6  1900-01-01 00:01:06.120    30.26     35.86  Womens 100M Breaststroke   
7  1900-01-01 00:01:06.230    31.06     35.17  Womens 100M Breaststroke   
8  1900-01-01 00:01:06.390    31.51     34.88  Womens 100M Breaststroke   
9  1900-01-01 00:01:06.560    31.39     35.17  Womens 100M Breaststroke   
10 1900-01-01 00:01:06.890    30.81     36.08  Womens 100M Breaststroke   
11 1900-01-01 00:01:06.900    31.02     35.88  Womens 100M Breaststroke   
12 1900-01-01 00:01:06.960    31.53     35.43  Womens 100M Breaststroke   
13 1900-01-01 00:01:07.200    31.40     35.80  Womens 100M Breaststroke   
14 1900-01-01 00:01:07.310    31.21     36.10  Womens 100M Breaststroke   
15 1900-01-01 00:01:07.480    31.29     36.19  Womens 100M Breaststroke   

         Round  
0   Semifinals  
1   Semifinals  
2   Semifinals  
3   Semifinals  
4   Semifinals  
5   Semifinals  
6   Semifinals  
7   Semifinals  
8   Semifinals  
9   Semifinals  
10  Semifinals  
11  Semifinals  
12  Semifinals  
13  Semifinals  
14  Semifinals  
15  Semifinals

Place Country                    Name Age    RT     Status  \
0       1     CHN                Jiayu XU  28  0.52  QUALIFIED   
1       2     ITA           Thomas CECCON  23  0.61  QUALIFIED   
2       3     FRA    Yohann NDOYE-BROUARD  23  0.55  QUALIFIED   
3       4     RSA           Pieter COETZE  20  0.67  QUALIFIED   
4       5     USA             Ryan MURPHY  29  0.52  QUALIFIED   
5       6     GRE      Apostolos CHRISTOU  27  0.58  QUALIFIED   
6       7     GBR           Oliver MORGAN  21  0.60  QUALIFIED   
7       8     ESP           Hugo GONZALEZ  25  0.64  QUALIFIED   
8       9     GRE  Evangelos MAKRYGIANNIS  24  0.56  QUALIFIED   
9      10     HUN              Hubert KOS  21  0.54  QUALIFIED   
10     11     USA        Hunter ARMSTRONG  23  0.66  QUALIFIED   
11     12     CZE         Miroslav KNEDLA  19  0.52  QUALIFIED   
12     13     POL          Ksawery MASIUK  19  0.61  QUALIFIED   
13     14     GBR       Jonathon MARSHALL  19  0.60  QUALIFIED   
14     15     FRA             Mewen TOMAC  22  0.61  QUALIFIED   
15     16     CAN           Blake TIERNEY  22  0.54  QUALIFIED   

                      Time 50 Split 100 Split                 Event  \
0  1900-01-01 00:00:52.020    24.89     27.13  Mens 100M Backstroke   
1  1900-01-01 00:00:52.580    25.46     27.12  Mens 100M Backstroke   
2  1900-01-01 00:00:52.630    25.65     26.98  Mens 100M Backstroke   
3  1900-01-01 00:00:52.630    25.40     27.23  Mens 100M Backstroke   
4  1900-01-01 00:00:52.720    25.49     27.23  Mens 100M Backstroke   
5  1900-01-01 00:00:52.770    25.42     27.35  Mens 100M Backstroke   
6  1900-01-01 00:00:52.850    25.69     27.16  Mens 100M Backstroke   
7  1900-01-01 00:00:52.950    25.51     27.44  Mens 100M Backstroke   
8  1900-01-01 00:00:52.970    25.48     27.49  Mens 100M Backstroke   
9  1900-01-01 00:00:52.980    25.87     27.11  Mens 100M Backstroke   
10 1900-01-01 00:00:53.110    25.53     27.58  Mens 100M Backstroke   
11 1900-01-01 00:00:53.440    25.37     28.07  Mens 100M Backstroke   
12 1900-01-01 00:00:53.440    25.53     27.91  Mens 100M Backstroke   
13 1900-01-01 00:00:53.460    25.89     27.57  Mens 100M Backstroke   
14 1900-01-01 00:00:53.630    25.81     27.82  Mens 100M Backstroke   
15 1900-01-01 00:00:53.710    26.06     27.65  Mens 100M Backstroke   

         Round  
0   Semifinals  
1   Semifinals  
2   Semifinals  
3   Semifinals  
4   Semifinals  
5   Semifinals  
6   Semifinals  
7   Semifinals  
8   Semifinals  
9   Semifinals  
10  Semifinals  
11  Semifinals  
12  Semifinals  
13  Semifinals  
14  Semifinals  
15  Semifinals

Place Country                Name Age    RT Status Time 50 Split 100 Split  \
0     1     ITA  Nicolo MARTINENGHI  24  0.63  59.03         27.34     31.69   
1     2     GBR          Adam PEATY  29  0.62  59.05         27.08     31.97   
2     2     USA            Nic FINK  31  0.69  59.05         27.45     31.60   
3     4     GER       Melvin IMOUDU  25  0.69  59.11         27.37     31.74   
4     5     GER     Lucas MATZERATH  24  0.72  59.30         27.56     31.74   
5     6     NED       Arno KAMMINGA  28  0.63  59.32         27.94     31.38   
6     7     CHN         Haiyang QIN  25  0.63  59.50         27.03     32.47   
7     8     NED      Caspar CORBEAU  23  0.68  59.98         27.91     32.07   

                    Event   Round  
0  Mens 100M Breaststroke  Finals  
1  Mens 100M Breaststroke  Finals  
2  Mens 100M Breaststroke  Finals  
3  Mens 100M Breaststroke  Finals  
4  Mens 100M Breaststroke  Finals  
5  Mens 100M Breaststroke  Finals  
6  Mens 100M Breaststroke  Finals  
7  Mens 100M Breaststroke  Finals

Place Country                 Name Age    RT   Status Time    50m   100m  \
0     1     FRA        Leon MARCHAND  22  0.66  4:02.95       25.12  29.20   
1     2     JPN  Tomoyuki MATSUSHITA  18  0.62  4:08.62       26.05  30.47   
2     3     USA        Carson FOSTER  22  0.68  4:08.66       25.93  29.71   
3     4     GBR       Max LITCHFIELD  29  0.65  4:08.85       26.41  29.82   
4     5     ITA     Alberto RAZZETTI  25  0.61  4:09.38       26.03  29.97   
5     6     NZL      Lewis CLAREBURT  25  0.71  4:10.44       26.05  29.98   
6     7     JPN           Daiya SETO  30  0.62  4:11.78       25.53  29.35   
7     8     GER       Cedric BUSSING  20  0.67  4:17.16       26.89  31.03   

    150m   200m   250m   300m   350m   400m                        Event  \
0  31.38  31.06  33.81  33.67  29.68  29.03  Mens 400M Individual Medley   
1  32.64  31.97  34.75  35.68  29.39  27.67  Mens 400M Individual Medley   
2  32.43  31.68  35.33  35.62  30.22  27.74  Mens 400M Individual Medley   
3  32.10  31.69  35.54  35.79  29.23  28.27  Mens 400M Individual Medley   
4  33.09  32.12  35.11  35.54  29.26  28.26  Mens 400M Individual Medley   
5  32.59  31.80  35.41  36.47  29.79  28.35  Mens 400M Individual Medley   
6  32.48  32.30  35.17  36.08  30.10  30.77  Mens 400M Individual Medley   
7  33.37  31.95  36.10  36.95  31.20  29.67  Mens 400M Individual Medley   

    Round  
0  Finals  
1  Finals  
2  Finals  
3  Finals  
4  Finals  
5  Finals  
6  Finals  
7  Finals

## Day 3

W400IM heats

W100 back heats

M800 free heats

W400IM finals

W100 breast finals

M200 free finals

W100 back semis

M100 back finals

W100 breast finals

W200 free finals


In [1]:
# parse W400IM heats
from utils import parse_800_heats

#w400IM_heats_df=parse_400_heats('W400IM_heats_results.csv', 'Womens 400M Individual Medley', 'Heats')
#w400IM_heats_df=sort_heats(w400IM_heats_df, topN=8)
#w400IM_heats_df.to_csv(outdir / 'W400IM_heats_results.csv', index=False)
#display(w400IM_heats_df)

#parse W100Back heats

#w100Back_heats_df=parse_100_heats('W100Back_heats_results.csv', 'Womens 100M Backstroke', 'Heats')
#w100Back_heats_df=sort_heats(w100Back_heats_df, topN=16)
#w100Back_heats_df.to_csv(outdir / 'W100Back_heats_results.csv', index=False)
#display(w100Back_heats_df)

#parse M800Free heats

#m800Free_heats_df=parse_800_heats('M800Free_heats_results.csv', 'Mens 800M Freestyle', 'Heats')
#m800Free_heats_df=sort_heats(m800Free_heats_df, topN=8)
#m800Free_heats_df.to_csv(outdir / 'M800Free_heats_results.csv', index=False)
#display(m800Free_heats_df)

['1\t6\t\n', 'AUTAUT\n', 'Felix AUBOECK\n', 'Felix\n', 'AUBOECK\n', '27\t0.80\t7:48.49\t-\t898\t\n', 'Distance\tSplit Time\tCumulative Time\n', ' 50m\n', '27.40\t27.40\n', ' 100m\n', '29.21\t56.61\n', ' 150m\n', '29.69\t01:26.30\n', ' 200m\n', '29.31\t01:55.61\n', ' 250m\n', '29.74\t02:25.35\n', ' 300m\n', '29.33\t02:54.68\n', ' 350m\n', '29.67\t03:24.35\n', ' 400m\n', '29.46\t03:53.81\n', ' 450m\n', '29.54\t04:23.35\n', ' 500m\n', '29.24\t04:52.59\n', ' 550m\n', '29.56\t05:22.15\n', ' 600m\n', '29.26\t05:51.41\n', ' 650m\n', '29.72\t06:21.13\n', ' 700m\n', '29.58\t06:50.71\n', ' 750m\n', '29.69\t07:20.40\n', ' 800m\n', '28.09\t07:48.49\n', '2\t3\t\n']
1	6	

27	0.80	7:48.49	-	898	

['ESPESP\n', 'Carlos GARACH BENITO\n', 'Carlos\n', 'GARACH BENITO\n', '20\t0.68\t7:50.07\t+01.58\t889\t\n', 'Distance\tSplit Time\tCumulative Time\n', ' 50m\n', '27.56\t27.56\n', ' 100m\n', '29.74\t57.30\n', ' 150m\n', '30.02\t01:27.32\n', ' 200m\n', '30.04\t01:57.36\n', ' 250m\n', '29.96\t02:27.32\n', ' 300

ValueError: not enough values to unpack (expected 3, got 1)

In [15]:
import pandas as pd
fname='M800Free_heats_results.csv'
results = []
with open(fname, 'r') as file:
    content = file.readlines()
records = [content[i:i + 39] for i in range(0, len(content), 39)]

results = []
for rec in records:
    #print (rec)
    print (rec[0])
    place = rec[0].strip().split()[0]
    country = rec[1].strip()
    country = country[0:3]
    name = rec[2].strip()
    print (rec[5])
    age_rt_status_points = rec[5].strip()

        
    age,rt,status_time = age_rt_status_points.split()[0:3]
        

    status = status_time[0:9]
    time = status_time[9::]

    splits = rec[7::]
    splits_times= splits[1::2]
    distances = splits[0::2]
    distances = [dist.strip() for dist in distances]

    split_times = [split.split('\t')[0] for split in splits_times]
    res = [place, country, name, age, rt, status, time]
    res.extend(split_times)
    res.extend(['M800Free', 'heats'])
    results.append(res)

        
        
    columns=['Place', 'Country', 'Name', 'Age', 'RT', 'Status', 'Time']
    columns=['Place', 'Country', 'Name', 'Age', 'RT', 'Status', 'Time']
    columns.extend(distances)
    columns.extend(['Event', 'Round'])
    df = pd.DataFrame(results, columns=columns)








['1\t6\t\n', 'AUTAUT\n', 'Felix AUBOECK\n', 'Felix\n', 'AUBOECK\n', '27\t0.80\t7:48.49\t-\t898\t\n', 'Distance\tSplit Time\tCumulative Time\n', ' 50m\n', '27.40\t27.40\n', ' 100m\n', '29.21\t56.61\n', ' 150m\n', '29.69\t01:26.30\n', ' 200m\n', '29.31\t01:55.61\n', ' 250m\n', '29.74\t02:25.35\n', ' 300m\n', '29.33\t02:54.68\n', ' 350m\n', '29.67\t03:24.35\n', ' 400m\n', '29.46\t03:53.81\n', ' 450m\n', '29.54\t04:23.35\n', ' 500m\n', '29.24\t04:52.59\n', ' 550m\n', '29.56\t05:22.15\n', ' 600m\n', '29.26\t05:51.41\n', ' 650m\n', '29.72\t06:21.13\n', ' 700m\n', '29.58\t06:50.71\n', ' 750m\n', '29.69\t07:20.40\n', ' 800m\n', '28.09\t07:48.49\n']
1	6	

27	0.80	7:48.49	-	898	

['2\t3\t\n', 'ESPESP\n', 'Carlos GARACH BENITO\n', 'Carlos\n', 'GARACH BENITO\n', '20\t0.68\t7:50.07\t+01.58\t889\t\n', 'Distance\tSplit Time\tCumulative Time\n', ' 50m\n', '27.56\t27.56\n', ' 100m\n', '29.74\t57.30\n', ' 150m\n', '30.02\t01:27.32\n', ' 200m\n', '30.04\t01:57.36\n', ' 250m\n', '29.96\t02:27.32\n', ' 300